lib/commonio.c

```diff
@@ -519,7 +519,7 @@ int commonio_open (struct commonio_db *db, int mode)
 	}
 
 	while (db->ops->fgets (buf, (int) buflen, db->fp) == buf) {
-		while (   ((cp = strrchr (buf, '\n')) == NULL)
+		while (   (strrchr (buf, '\n') == NULL)
```

Comment:
This whole loop seems like some `getline(3)` implementation.  If so, we could just call it.  Could you please check that?